org.mixedrealitytoolkit.input/Interactors/Gaze/TrackedPoseDriverWithFallback.cs

```diff
@@ -0,0 +1,116 @@
+        /// </summary>
+        public InputActionProperty FallbackTrackingStateAction => fallbackTrackingStateAction;
+
+        #endregion Fallback action values
+
+        #region TrackedPoseDriver Overrides 
+        /// <inheritdoc />
+        protected override void PerformUpdate()
+        {
+            base.PerformUpdate();
+
+            if (trackingStateInput.action == null)
+            {
+                Debug.LogWarning("TrackedPoseDriverWithFallback.trackingStateInput.action is null, no fallback will be used.");
+                return;
+            }
+
+            var hasPositionAction = positionAction != null;
+            var hasPositionFallbackAction = fallbackPositionAction != null;
+
+            var hasRotationAction = rotationAction != null;
+            var hasRotationFallbackAction = fallbackRotationAction != null;
+
+            InputTrackingState inputTrackingState = (InputTrackingState)trackingStateInput.action.ReadValue<int>();
+
+            // If default InputTrackingState does not have position and rotation data, use fallback if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && !inputTrackingState.HasFlag(InputTrackingState.Rotation) && FallbackTrackingStateAction.action != null)
+            {
+                inputTrackingState = (InputTrackingState)FallbackTrackingStateAction.action.ReadValue<int>();
+            }
+
+            bool neededToGetFallbackData = false;
+            Vector3 position = Vector3.zero;
+            Quaternion rotation = Quaternion.identity;
+
+            // If no position data then use the data from the fallback action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && hasPositionFallbackAction)
+            {
+                neededToGetFallbackData = true;
+                position = fallbackPositionAction.action.ReadValue<Vector3>();
+            }
+            else if (hasPositionAction)
+            {
+                position = positionAction.ReadValue<Vector3>(); //get a hold on the position data in case orientation data is obtained from fallback action
+            }
+
+            // If no rotation data then use the data from the fallback action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Rotation) && hasRotationFallbackAction)
+            {
+                neededToGetFallbackData = true;
+                rotation = fallbackRotationAction.action.ReadValue<Quaternion>();
+            }
+            else if (hasRotationAction)
+            {
+                rotation = rotationAction.ReadValue<Quaternion>(); //get a hold on the rotation data in case position data was obtained from fallback action
+            }
+
+            if (neededToGetFallbackData) //because either position, rotation, or both data were obtained from fallback actions
+            {
+                SetLocalTransform(position, rotation);
```

Comment:
> I took a closer look at this `SetLocalTransform`, I actually think this is not going to work if the primary input is not tracked. See line 585 of TrackedPoseDriver.cs

That's why I added this condition in which if the primary input is not tracked then the InputTrackingState is ready from the FallbackTrackingStateAction if it exists:

<img width="648" alt="image" src="https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/assets/84108471/049fdf78-4d95-4b16-b1b5-7f0dac085b4c">
 
> So I think we need to make our own `SetLocalTransformFromFallback` method, and call this instead:
> 
> ```cs
> protected virtual void SetLocalTransformFromFallback(Vector3 newPosition, Quaternion newRotation)
> {
>     var positionValid = this.IgnoreTrackingState || (m_CurrentFallbackTrackingState & TrackingStates.Position) != 0;
>     var rotationValid = this.IgnoreTrackingState || (m_CurrentFallbackTrackingState & TrackingStates.Rotation) != 0;
> 
> #if HAS_SET_LOCAL_POSITION_AND_ROTATION
>     if (this.TrackingType == TrackingType.RotationAndPosition && rotationValid && positionValid)
>     {
>         transform.SetLocalPositionAndRotation(newPosition, newRotation);
>         return;
>     }
> #endif
> 
>     if (rotationValid &&
>         (this.TrackingType == TrackingType.RotationAndPosition ||
>          this.TrackingType == TrackingType.RotationOnly))
>     {
>         transform.localRotation = newRotation;
>     }
> 
>     if (positionValid &&
>         (this.TrackingType == TrackingType.RotationAndPosition ||
>          this.TrackingType == TrackingType.PositionOnly))
>     {
>         transform.localPosition = newPosition;
>     }
> }
> ```

I think the condition I mentioned above does cover the case you are mentioning.  Could you please confirm whether it covers or not the case you mention?
Please let me know so that I can either revert or leave commit https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/746/commits/5ccb160ceddb922e1c03e7921e2010095954ec4e in which I implement your suggestion.